### PR TITLE
Replaces holo tape with holo barrier 

### LINKED ISSUE #7961: Holotape has many bugs
@paprka
* It can be destroyed with ranged and melee weapons but neither give any feedback whatsoever (no animation, no message).
* Clicking an airlock to tape it also activates the airlock. With windows the mob attacks it. 
* Trying to tape an airlock during the holotape cooldown gives no feedback.
* You can spamclick an airlock to place multiple tape on one airlock.
* simple animal and slimes cannot pass holotapes.
* simple animal, slimes, humanoid alien, and larvas cannot attack holotapes.
* You can place a tape line over a tape line (it won't actually be placed but the messages you get at start and finish are the standard ones)
* You can place tape over windows but the tape is then inaccessible and can't be removed/attacked w/o first destroying the window.
* Multiple mob can be on the same tape tile, they don't swap places or push each other, they basically stack like ghosts.
* Trying to move (via push, pull or grab) anything that isn't a mob onto the tape causes a runtime error (shown below).
* A mob on top of a holotape can't be hit a projectile (It only hits the tape), but the mob can fire projectiles at his enemies.

Usability issues: 
* Can't delete the holotapes with the tape projector (like the janitor's sign projector). 
* No clear way to cancel taping after starting it, you have to move the projector or click it when it won't actually place the tape. 

Consistency issue:
* You can make a tape line pass through a taped airlock, you can make two tape line cross, but you can't tape an airlock if there's a tape line on its tile.
* Placing overlapping tape lines (both vertical or both horizontal) is inconsistent. if you start your second line on an empty tile and finish it on a tile where the first tapeline is then the empty tiles will get a tape. But if you started on an occupied tile and finished on an empty one then no tape is placed, and in both cases you get the same standard successful messages.

Runtime:

runtime error: undefined variable /obj/structure/computerframe/var/m_intent
proc name: Bumped (/obj/item/holotape/Bumped)
  source file: holotape.dm,157
  usr: 0
  src: the security holotape (/obj/item/holotape/security)
  call stack:
the security holotape (/obj/item/holotape/security): Bumped(the computer-frame (/obj/structure/computerframe))
the computer-frame (/obj/structure/computerframe): Bump(the security holotape (/obj/item/holotape/security), 1)
the floor (191,132,1) (/turf/simulated/floor/plasteel): Enter(the computer-frame (/obj/structure/computerframe), the floor (191,131,1) (/turf/simulated/floor/plasteel))
the computer-frame (/obj/structure/computerframe): Move(the floor (191,132,1) (/turf/simulated/floor/plasteel), 1)
Alaina Lloyd (/mob/living/carbon/human): PushAM(the computer-frame (/obj/structure/computerframe))
Alaina Lloyd (/mob/living/carbon/human): Bump(the computer-frame (/obj/structure/computerframe), 1)
the floor (191,131,1) (/turf/simulated/floor/plasteel): Enter(Alaina Lloyd (/mob/living/carbon/human), the floor (191,130,1) (/turf/simulated/floor/plasteel))
Alaina Lloyd (/mob/living/carbon/human): Move(the floor (191,131,1) (/turf/simulated/floor/plasteel), 1)
Alaina Lloyd (/mob/living/carbon/human): Move(the floor (191,131,1) (/turf/simulated/floor/plasteel), 1)
Alaina Lloyd (/mob/living/carbon/human): Move(the floor (191,131,1) (/turf/simulated/floor/plasteel), 1)
Alaina Lloyd (/mob/living/carbon/human): Move(the floor (191,131,1) (/turf/simulated/floor/plasteel), 1)
Phil235 (/client): Move(the floor (191,131,1) (/turf/simulated/floor/plasteel), 1)

### PR DESCRIPTION
**Pull request by phil235**
Replaces holo tape and holo tape projector with holo barrier and holo barrier projector (children of the janitor's holosign creator).

All holosign_creators' code is in holosign_creator.dm and their icons in device.dmi
All holo signs themselves are in effects.dmi

The holo barrier is better than holotape in every way.
- It can be destroyed by all mob attacks and mech attacks, and projectiles.
- it's harder to abuse, you can't put more than one barrier on a floor tile
- you can't  put a barrier if there's something dense on the tile.
- Placing a holobarrier takes 3 seconds so you can't use it during chases.
- Holobarrier are placed one at a time so you can't quickly block a hallway.
- The barrier projector has a sign limit of 6 barriers to avoid abuse.
- You can't place barrier on windows. 
- Mobs with PASSTABLE, PASSGRILLE can pass through the barrier unrestricted, other mobs have to walk. 
- Things that pass through glass also pass the barrier.
- Projectile and thrown items pass over the barrier. 
- You can shoot the barrier. 
- The damage dealt to the barrier by hand are no longer random.

These changes also apply to the janitor's holosign projector (except creating a wet floor holosign is still instantaneous). The holo wet sign can be destroyed by attacks, and placing one does a click sound.

Shitty coder sprites courtesy of yours truly.

Fixes #7961

:cl: phil235
rscadd: Holo tape is replaced by holo barriers.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/portals.dm [hunk 1]
```diff
@@ -1,5 +1,6 @@
 
 /obj/effect
+	icon = 'icons/effects/effects.dmi'
```
Comment (phil235) [inline code/game/objects/effects/portals.dm]: so we don't have to add that line to more than half of /obj/effect's children.

File: code/game/objects/items/holotape.dm [hunk 2]
```diff
@@ -1,254 +0,0 @@
...
-
-	if(M.has_unlimited_silicon_privilege)
-		var/turf/T = get_turf(src)
-		M.loc = T
-
-/obj/item/holotape/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
-	if(!density) return 1
-	if(air_group || (height==0)) return 1
-
-	if((mover.flags & PASSGLASS || istype(mover, /obj/effect/meteor) || mover.throwing == 1) )
```
Comment (phil235) [inline code/game/objects/items/holotape.dm]: This line man.... PASSGLASS is a bit used by pass_flags, not flags. And a specific check for meteors but nothing for any projectile at all?

Comment (KazeEspada) [inline code/game/objects/items/holotape.dm]: >Meteors go through. The one thing that an engineer would be most worried about.

Comment (Cheridan) [inline code/game/objects/items/holotape.dm]: I'd bet anything there was some guy on Bay meticulously placing down holotape all round to block all meteors so they had to add this :P

File: code/game/objects/items/weapons/holosign_creator.dm [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+		return 1
+	if(iscarbon(mover))
+		var/mob/living/carbon/C = mover
+		if(C.m_intent == "walk")
+			return 1
+
+/obj/effect/overlay/holograph/barrier/engineering
+	icon_state = "holosign_engi"
+
+/obj/item/weapon/caution
```
Comment (KazeEspada) [inline code/game/objects/items/weapons/holosign_creator.dm]: Why is this here? Its not holographic?

Comment (phil235) [inline code/game/objects/items/weapons/holosign_creator.dm]: It was in the file janisigns.dm that I deleted and replaced by holosign_creator.dm
I didn't know where else to put it.

Comment (KazeEspada) [inline code/game/objects/items/weapons/holosign_creator.dm]: Miscenlaneous weapons?

**Comment by Alek2ander** [comment]
Can mechs pass barriers?

**Comment by KazeEspada** [comment]
No, they have to smash them. They are guarenteed to one hit smash them.

File: code/game/objects/items/weapons/holosign_creator.dm [hunk 4]
```diff
@@ -0,0 +1,189 @@
+/obj/item/weapon/holosign_creator
+	name = "holographic sign projector"
+	desc = "A handy-dandy hologaphic projector that displays a janitorial sign."
```
Comment (tkdrg) [inline code/game/objects/items/weapons/holosign_creator.dm]: hologRaphic

File: code/game/objects/items/weapons/holosign_creator.dm [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+		var/obj/effect/overlay/holograph/H = locate(holosign_type) in T
+		if(H)
+			user << "<span class='notice'>You use [src] to deactivate [H].</span>"
+			signs.Remove(H)
+			qdel(H)
+		else
+			if(!T.density) //can't put holograms on a tile that has dense stuff
+				for(var/atom/movable/AM in T)
+					if(AM.density)
+						return
```
Comment (tkdrg) [inline code/game/objects/items/weapons/holosign_creator.dm]: you can replace this boilerplate with the `if(is_blocked_turf(T))` helper

File: code/game/objects/items/weapons/holosign_creator.dm [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+						holocreator_busy = 1
+						if(!do_after(user, creation_time, target = target))
+							holocreator_busy = 0
+							return
+						holocreator_busy = 0
+						if(signs.len >= max_signs || T.density)
+							return
+						for(var/atom/movable/AM in T) //don't try to sneak dense stuff on our tile during the wait.
+							if(AM.density)
+								return
```
Comment (tkdrg) [inline code/game/objects/items/weapons/holosign_creator.dm]: can use the same helper here

**Comment by tkdrg** [comment]
@phil235 looks fine to me but conflicts
